src/main/java/net/sf/jabref/model/entry/InternalBibtexFields.java

```diff
@@ -305,17 +273,29 @@ private InternalBibtexFields(boolean serializeSpecialFields) {
             add(field);
         }
 
-        // collect all public fields for the PUBLIC_FIELDS array
-        for (BibtexSingleField sField : fieldSet.values()) {
-            if (!sField.isPrivate()) {
-                PUBLIC_FIELDS.add(sField.getFieldName());
-                // or export the complete BibtexSingleField ?
-                // BibtexSingleField.toString() { return fieldname ; }
-            }
-        }
+    }
 
-        // sort the entries
-        Collections.sort(PUBLIC_FIELDS);
+
+    public static void updateTimeStampField(String timeStampFieldName) {
```

Comment:
I would propose to switch to a "normal class" without using the singleton pattern. The constructor takes some preferences as input and then initializes the fields accordingly. In this way you don't need these static update methods.